### ISSUE TITLE
Easy Application layer DDOS

### ISSUE THREAD

**Issue by picatextra** [X-CASH-official/xcash-dpops#17]
# 🐛 BUG REPORT 

I can make a delegate unresponsive by sending non http data

**Expected Behavior**

The delegates communications should not be blocked

**Current Behavior**

receive_data() doesn't handle well when data is received without an end string
recvfrom is keep called until the timeout .

The DDOS attack is easy because there is no need to send huge data,
the attacker just need to send enough request so that the timeout sum  if more than the block produce interval. So that the delegate can never communicate with others in time.

**Possible Solution**

When data was received on the socket , the next call to recvfrom will return -1, then we should exit the loop.

**Steps to Reproduce (for bugs)**

Send non http data to the delegate's 18283 port

Javascript example :

```
const Net = require('net');
const client = new Net.Socket();

client.connect({ port: 18283, host: 'delegate.xxx' }, function() {    
    console.log('Connection established with the server.');    
    client.write("Hello");
})
.on('end', function() {
    console.log('Connection ended');
});
```

**Comment by zachhildreth** [comment]
Since this issue is related to issue #16 please refer to that issue for the bounty, discussions and code fixes

**Comment by picatextra** [comment]
Here is an other possible solution,

for example if i send a simple "hello" to my TCP socket :
the recvfrom will return 6  (`h,e,l,l,o,\0`)
so if the last char received is `\0` , we can exit the loop , 

something link this :
```
count = recvfrom(...);
if(buffer[count-1] =='\0') break;
```

**Comment by plbgnt** [comment]
Fixed with #16
